tasks/unit-tests/update_go_tests.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from ..update_go import (
+    PATTERN_MAJOR_MINOR,
+    PATTERN_MAJOR_MINOR_BUGFIX,
+    _get_major_minor_version,
+    _get_pattern,
+    _update_go_mods,
+    _update_references,
+)
+
+
+class TestUpdateGo(unittest.TestCase):
+    def test_get_minor_version(self):
+        self.assertEqual(_get_major_minor_version("1.2.3"), "1.2")
+
+    def test_get_pattern(self):
+        self.assertEqual(_get_pattern("p+e", "p.st", is_bugfix=True), rf'(p\+e){PATTERN_MAJOR_MINOR_BUGFIX}(p\.st)')
+        self.assertEqual(_get_pattern("p(re)", "p*st", is_bugfix=False), rf'(p\(re\)){PATTERN_MAJOR_MINOR}(p\*st)')
+
+    def test_file_patterns(self):
+        _update_references(warn=False, version="1.2.3", dry_run=True)
+
+    def test_go_mods(self):
+        _update_go_mods(warn=False, minor="1.2", include_otel_modules=True, dry_run=True)
+
```

Comment:
These two tests confirm that calling the method does not raise an exception, right? 

Should we modify the methods when using `dry_run=True,` so we can check which files would have been modified?

Reply:
> These two tests confirm that calling the method does not raise an exception, right?

Yes, the thing is that the task raises an exception if it can't find all expected patterns in the files, so if it doesn't raise an exception you already know that everything is setup as expected (all files still exist, and the patterns in the files still exist too).

Reply:
The only thing `dry_run=True` changes is avoid actually performing the replacement of Go version, only check that the patterns exist